tests/integration/test_types_integration.py

```diff
@@ -0,0 +1,208 @@
+        .add_field(sql="cast('2001-08-22 01:23:45.123' AS TIMESTAMP(3))",
+                   python=datetime.datetime(2001, 8, 22, 1, 23, 45, 123000)) \
+        .add_field(sql="cast(null AS TIMESTAMP(3))", python=None) \
+        .add_field(sql="cast('2001-08-22 01:23:45.123456' AS TIMESTAMP(6))",
+                   python=datetime.datetime(2001, 8, 22, 1, 23, 45, 123000)) \
+        .add_field(sql="cast(null AS TIMESTAMP(6))", python=None) \
+        .add_field(sql="cast('2001-08-22 01:23:45.123456789' AS TIMESTAMP(9))",
+                   python=datetime.datetime(2001, 8, 22, 1, 23, 45, 123000)) \
+        .add_field(sql="cast(null AS TIMESTAMP(9))", python=None) \
+        .add_field(sql="cast('2001-08-22 01:23:45.123456789123' AS TIMESTAMP(12))",
+                   python=datetime.datetime(2001, 8, 22, 1, 23, 45, 123000)) \
+        .add_field(sql="cast(null AS TIMESTAMP(12))", python=None) \
+        .add_field(sql="TIMESTAMP '2001-08-22 01:23:45.123 -08:00'",
+                   python=datetime.datetime(2001, 8, 22, 1, 23, 45, 123000, tzinfo=the_tz)) \
+        .add_field(sql="cast(null AS TIMESTAMP WITH TIME ZONE)", python=None) \
+        .add_field(sql="cast('2001-08-22 01:23:45 -08:00' AS TIMESTAMP(0) WITH TIME ZONE)",
+                   python=datetime.datetime(2001, 8, 22, 1, 23, 45, tzinfo=the_tz)) \
+        .add_field(sql="cast(null AS TIMESTAMP(0) WITH TIME ZONE)", python=None) \
+        .add_field(sql="cast('2001-08-22 01:23:45.123 -08:00' AS TIMESTAMP(3) WITH TIME ZONE)",
+                   python=datetime.datetime(2001, 8, 22, 1, 23, 45, 123000, tzinfo=the_tz)) \
+        .add_field(sql="cast(null AS TIMESTAMP(3) WITH TIME ZONE)", python=None) \
+        .add_field(sql="cast('2001-08-22 01:23:45.123456 -08:00' AS TIMESTAMP(6) WITH TIME ZONE)",
+                   python=datetime.datetime(2001, 8, 22, 1, 23, 45, 123000, tzinfo=the_tz)) \
+        .add_field(sql="cast(null AS TIMESTAMP(6) WITH TIME ZONE)", python=None) \
+        .add_field(sql="cast('2001-08-22 01:23:45.123456789 -08:00' AS TIMESTAMP(9) WITH TIME ZONE)",
+                   python=datetime.datetime(2001, 8, 22, 1, 23, 45, 123000, tzinfo=the_tz)) \
+        .add_field(sql="cast(null AS TIMESTAMP(9) WITH TIME ZONE)", python=None) \
+        .add_field(sql="cast('2001-08-22 01:23:45.123456789123 -08:00' AS TIMESTAMP(12) WITH TIME ZONE)",
+                   python=datetime.datetime(2001, 8, 22, 1, 23, 45, 123000, tzinfo=the_tz)) \
+        .add_field(sql="cast(null AS TIMESTAMP(12) WITH TIME ZONE)", python=None) \
+        .execute()
+
+
+def test_misc_types(trino_connection):
+    SqlTest(trino_connection) \
+        .add_field(sql="INTERVAL '3' MONTH", python='0-3') \
+        .add_field(sql="cast(null AS INTERVAL YEAR TO MONTH)", python=None) \
+        .add_field(sql="INTERVAL '2' DAY", python='2 00:00:00.000') \
+        .add_field(sql="cast(null AS INTERVAL DAY TO SECOND)", python=None) \
```

Comment:
Add a negative sign case for `INTERVAL` type.

Reply:
Done